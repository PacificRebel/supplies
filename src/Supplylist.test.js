import React from 'react';
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import renderer from 'react-test-renderer';

import SupplyList from './SupplyList'

test('user can add a supply', () => {
  render(<SupplyList />);

  const textBox = screen.getByRole("textbox");
  userEvent.type(textBox, "A test supply")
  userEvent.click(screen.getByText("Add supply to list"))
  const supply = screen.getByText("A test supply");

  expect(supply).toBeInTheDocument();

})

  test('renders correctly when there are no supplies', () => {
    const tree = renderer.create(<SupplyList />).toJSON();
    expect(tree).toMatchSnapshot();
  })
