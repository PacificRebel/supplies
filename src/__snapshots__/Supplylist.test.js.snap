// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no supplies 1`] = `
<div
  className="SupplyList"
>
  <form
    onSubmit={[Function]}
  >
    <label>
      Add supply: 
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <input
      type="submit"
      value="Add supply to list"
    />
  </form>
</div>
`;
